src/pages/UserPosts.js

```diff
@@ -21,22 +23,78 @@ export default function UserPosts() {
   const [disabled, setDisabled] = useState(false);
   const [follows, setFollows] = useState([]);
   const [following, setFollowing] = useState(false);
+  const [trasnfer, setTrasnfer] = useState(false);
+  let higher = Number.POSITIVE_INFINITY;
+  const [postsIds, setPostsIds] = useState([]);
+  const [firstPostId, setFirstPostId] = useState(0);
+  const [pageNumber, setPageNumber] = useState(0);
+  const [newUserPosts, setNewUserPosts] = useState([]);
+  const [hasMore, setHasMore] = useState(true);
 
-  useEffect(() => {
-    getUserPosts(id, { token: userData.token }).then((r) =>
-      setUserPosts(r.data.posts)
-    );
-    getFollows(userData.token).then((r) => setFollows(r.data.users));
-  }, [following, id]);
+  function postRepost(post) {
+    if(post.repostId) {
+      return post.repostId;
+    } else {
+      return post.id;
+    }
+  }
```

Comment:
```
  function postRepost(post) {
    return post.repostId ? post.repostId : post.id;
  }
```

Reply:
I've put the this sentence directly inside useEffect callback.